python/cudf/cudf/core/multiindex.py

```diff
@@ -392,13 +393,93 @@ def _popn(self, n):
         return result
 
     def __repr__(self):
-        return (
-            "MultiIndex(levels="
-            + str(self.levels)
-            + ",\ncodes="
-            + str(self.codes)
-            + ")"
+        max_seq_items = get_option("display.max_seq_items") or len(self)
+
+        if len(self) > max_seq_items:
+            n = int(max_seq_items / 2) + 1
+            top = self[:n]
+            bottom = self[-n:]
+            preprocess = cudf.concat([top, bottom])
```

Comment:
might be better to use `take` to just gather all the indices at once, rather than getting the top and bottom and then concatenating them back together